File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/exoop/ShapeMain.java
package asiantech.internship.summer.java_core.exoop;

import java.util.ArrayList;
import java.util.List;
import java.util.logging.Logger;

public class ShapeMain {
    public static void main(String[] args) {
        Logger log = Logger.getLogger(ShapeMain.class.getName());
        Shape rectangle = new Rectangle(3, 4);
        Shape square = new Square(5);
        Shape circle = new Circle(3);

        List<Shape> shapes = new ArrayList<>();
        shapes.add(circle);
        shapes.add(square);
        shapes.add(rectangle);

        for (Shape shape : shapes) {
            log.info(shape.print() + ": " + shape.perimeter() + " - " + shape.area());
        }
    }
}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/exoop/ShapeMain.kt
package asiantech.internship.summer.kotlin.exoop

import java.util.ArrayList

object ShapeMain {
    @JvmStatic
    fun main(args: Array<String>) {
        val rectangle = Rectangle(3, 4)
        val square = Square(5)
        val circle = Circle(3)

        val shapes = ArrayList<Shape>()
        shapes.add(circle)
        shapes.add(square)
        shapes.add(rectangle)

        for (shape in shapes) {
            println(shape.print() + ": " + shape.perimeter() + " - " + shape.area())
        }
    }
}
